src/components/hosts/HostDialogsContext.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { Host, Inventory } from '../../api/types';
+import dialogsReducer, {
+  openDialog as openDialogAction,
+  closeDialog as closeDialogAction,
+} from '../../features/dialogs/dialogsSlice';
+
+type HostIdAndHostname = {
+  hostId: Host['id'];
+  hostname: string;
+};
+
+type EditHostProps = {
+  host: Host;
+  inventory: Inventory;
+};
+
+type DialogsDataTypes = {
+  eventsDialog: HostIdAndHostname;
+  editHostDialog: EditHostProps;
+  deleteHostDialog: HostIdAndHostname;
+  resetHostDialog: HostIdAndHostname;
+  additionalNTPSourcesDialog: void;
+};
+
+type DialogId =
+  | 'eventsDialog'
+  | 'editHostDialog'
+  | 'deleteHostDialog'
+  | 'resetHostDialog'
+  | 'additionalNTPSourcesDialog';
+
+export type HostDialogsContextType = {
+  [key in DialogId]: {
+    isOpen: boolean;
+    open: (data: DialogsDataTypes[key]) => void;
+    close: () => void;
+    data?: DialogsDataTypes[key];
+  };
+};
+
+const dialogIds: DialogId[] = [
+  'eventsDialog',
+  'editHostDialog',
+  'deleteHostDialog',
+  'resetHostDialog',
+  'additionalNTPSourcesDialog',
+];
+
+const HostDialogsContext = React.createContext<HostDialogsContextType | undefined>(undefined);
+
+const HostDialogsContextProvider: React.FC = ({ children }) => {
+  const [dialogsState, dispatchDialogsAction] = React.useReducer(dialogsReducer, {});
+
+  function getOpenDialog<DataType>(dialogId: string) {
+    return (data: DataType) => dispatchDialogsAction(openDialogAction({ dialogId, data }));
+  }
+
+  const getCloseDialog = (dialogId: string) => () =>
+    dispatchDialogsAction(closeDialogAction({ dialogId }));
+
+  const context = dialogIds.reduce((context, dialogId) => {
+    context[dialogId] = {
+      isOpen: !!dialogsState[dialogId],
+      open: (data: DialogsDataTypes[typeof dialogId]) =>
+        getOpenDialog<DialogsDataTypes[typeof dialogId]>(dialogId)(data),
+      close: () => getCloseDialog(dialogId)(),
+      data: dialogsState[dialogId],
+    };
+    return context;
+  }, {});
+
+  return (
+    <HostDialogsContext.Provider value={context as HostDialogsContextType}>
```

Comment:
nit: it would be safer for future maintenance to ensure that `context` is already of `HostDialogsContextType` type, especially since it is calculated.

Reply:
I can do the type conversion right after the reduce. Would that be sufficient? If I add the type to context definition, the initial value in `reduce` (`{}`) is required to match the `HostDialogsContextType` which is not what we want to do.

Reply:
That would not bring the intended help.
We can add `?` at line 34, that is safe enough since we are passing through all `dialogIds` here.
Or provide an inital default prior calling `reduce`.
Or do something like that round the line 63:
```
const myTemp: NewTypeForSingleDialogData = {
  isOpen: !!dialogsState[dialogId],
  ...
};
context[dialogId] = myTemp;
```

Reply:
I mean ^^^ as a code quality improvement, it's not a must.
The computation of context is not the simplest one and future maintenance of it could lead to issues which can be discovered by type-checking.